### PR TITLE
server: support partitioned table for table scatter (#17624)

### LINKED ISSUE #17623: Support partitioned table for table scatter API
## Development Task
```
curl  http://{TiDBIP}:10080/tables/{db}/{table}/scatter
```
the API does not support partitioned tables.

### PR DESCRIPTION
**Pull request by nolouch**
### What problem does this PR solve?
support partitioned table for table scatter
Issue Number: close #17623
pick: https://github.com/pingcap/tidb/pull/17624
### What is changed and how it works?

How it Works:
 - use psychic table id.

### Release note 

- Fix the partition table issue in table scatter API.

### CODE REVIEW

File: docs/tidb_http_api.md [hunk 1]
```diff
@@ -261,16 +261,18 @@ timezone.*
 1. Scatter regions of the specified table, add a `scatter-range` scheduler for the PD and the range is same as the table range.
 
     ```shell
-    curl -X POST http://{TiDBIP}:10080/tables/{db}/{table}/scatter
+    curl http://{TiDBIP}:10080/tables/{db}/{table}/scatter
     ```
+    *Hint: On a partitioned table, use the `table(partition)` pattern as the table name, `test(p1)` for example:*
```
Comment (zimulala) [inline docs/tidb_http_api.md:266]:
```suggestion
    *Hint: On a partitioned table, use the `table(partition)` pattern as the table name, `test(p1)` for example.*
```

File: docs/tidb_http_api.md [hunk 2]
```diff
@@ -261,16 +261,18 @@ timezone.*
...
 
     **Note**: The `scatter-range` scheduler may conflict with the global scheduler, do not use it for long periods on the larger table.
 
 1. Stop scatter the regions, disable the `scatter-range` scheduler for the specified table.
 
     ```shell
-    curl -X POST http://{TiDBIP}:10080/tables/{db}/{table}/stop-scatter
+    curl http://{TiDBIP}:10080/tables/{db}/{table}/stop-scatter
     ```
+    *Hint: On a partitioned table, use the `table(partition)` pattern as the table name, `test(p1)` for example:*
```
Comment (zimulala) [inline docs/tidb_http_api.md:275]:
```suggestion
    *Hint: On a partitioned table, use the `table(partition)` pattern as the table name, `test(p1)` for example.*
```

**Comment by nolouch** [comment]
PTAL @zimulala

**Pull request merged**